addons/mail/static/src/models/message/message.js

```diff
@@ -442,6 +442,33 @@ function factory(dependencies) {
                 this.originThread.isModeratedByCurrentPartner
             );
         }
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsSubjectEqualThreadName() {
+            if (!this.subject || !this.originThreadName) {
+                return false;
+            }
+
+            const threadName = this.originThreadName.toLowerCase().trim();
+            let prefix = ['re:', 'fw:', 'fwd:'];
+            let startingSubject = this.subject.toLowerCase();
+            let cleanedSubject;
+            while (startingSubject !== cleanedSubject) {
+                cleanedSubject = startingSubject;
+                for (const pref of prefix) {
+                    if (startingSubject.startsWith(pref)) {
+                        startingSubject = startingSubject.replace(pref, '').trim();
```

Comment:
`startingSubject` & `cleanedSubject` names have been inverted I think.
`cleanedSubject` should be the one updated here, and `startingSubject = cleanedSubject` at the start of the while